pkg/relayer/proxy/server_builder.go

```diff
@@ -26,10 +28,30 @@ func (rp *relayerProxy) BuildProvidedServices(ctx context.Context) error {
 		return err
 	}
 
-	// Get the supplier's advertised information from the blockchain
-	supplier, err := rp.supplierQuerier.GetSupplier(ctx, supplierAddress.String())
-	if err != nil {
-		return err
+	var supplier sharedtypes.Supplier
+
+	// Wait for the supplier to be staked before building the relay servers.
```

Comment:
OPTIONAL NIT: What do you think of moving this business logic into a helper and updating the comment that it's a while loop needed to get back a supplier?